src/core/dbus-execute.c

```diff
@@ -1826,6 +1862,33 @@ int bus_exec_context_set_transient_property(
         if (streq(name, "LogRateLimitBurst"))
                 return bus_set_transient_unsigned(u, name, &c->log_ratelimit_burst, message, flags, error);
 
+        if (streq(name, "LogFilterPatterns")) {
+                char *pattern;
+                bool is_allowlist;
+
+                r = sd_bus_message_enter_container(message, 'a', "(bs)");
+                if (r < 0)
+                        return r;
+
+                while ((r = sd_bus_message_read(message, "(bs)", &is_allowlist, &pattern)) > 0) {
+                        r = exec_store_log_filter_pattern(c, pattern, is_allowlist);
+                        if (r < 0)
+                                return r;
+                }
+
+                SET_FOREACH(pattern, c->log_filter_allowed_patterns)
+                        unit_write_settingf(u, flags, name, "%s=%s", name, pattern);
+
+                SET_FOREACH(pattern, c->log_filter_denied_patterns)
+                        unit_write_settingf(u, flags, name, "%s=~%s", name, pattern);
```

Comment:
Please do not write unit file unconditionally, this must be done only when `UNIT_WRITE_FLAGS_NOOP(flags))` is false.

Please see other settings in the function. The code should be something like the following:
```c
        if (streq(name, "LogFilterPatterns")) {
                _cleanup_set_free_ Set *allow_list = NULL, *deny_list = NULL;
                char *pattern;
                bool is_allowlist;

                r = sd_bus_message_enter_container(message, 'a', "(bs)");
                if (r < 0)
                        return r;

                while ((r = sd_bus_message_read(message, "(bs)", &is_allowlist, &pattern)) > 0) {
                        r = exec_store_log_filter_pattern(c, pattern, is_allowlist ? &allow_list : &deny_list);
                        if (r < 0)
                                return r;
                if (r < 0)
                        return r;

                r = sd_bus_message_exit_container(message);
                if (r < 0)
                        return r;

                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
                        if (set_isempty(allow_list) && set_isempty(deny_list)) {
                                /* Empty assignment resets the lists. */
                                c->log_filter_allowed_patterns = set_free(c->log_filter_allowed_patterns);
                                c->log_filter_denied_patterns = set_free(c->log_filter_denied_patterns);
                                unit_write_settingf(u, flags, name, "%s=", name);
                        } else {
                                 /* Merge with the existing lists. */
                                 SET_FOREACH(...)
                                         r = set_put_strdup(...);
                                         if (r < 0)
                                                 return r;
                                 /* Then, write unit file. */
                                 SET_FOREACH(...)
                                         unit_write_settingf(...);
                        }
                }
        }
```

Reply:
Thanks for taking the time to write this example.